samza-test/src/main/java/org/apache/samza/test/framework/TestRunner.java

```diff
@@ -364,4 +365,26 @@ public void run(Duration timeout) {
         .collect(Collectors.toMap(entry -> entry.getKey().getPartition().getPartitionId(),
             entry -> entry.getValue().stream().map(e -> (T) e.getMessage()).collect(Collectors.toList())));
   }
+
+  private TaskFactory createTaskFactory() {
```

Comment:
Isn't this the same method as TaskFactoryUtil#getTaskFactoryFromConfig?

Reply:
I think this allows the non class-loading methods in test environment to create the TaskFactory from classes like anonymous classes. Not sure whether that's the intention but I would prefer to leave it as part of the cleanup in test framework as in SAMZA-1818.